File: python/leetcode/169.py
class Solution(object):
    def majorityElement(self, nums):
        cnts, res = 0, 0
        for num in nums:
            if cnts == 0:
                res = num
                cnts = 1
            else:
                cnts += 1 if res == num else -1
        return res

File: cpp/leetcode/169.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int major, counts = 0, n = nums.size();
        for (int i = 0; i < n; i++) {
            if (counts == 0) {
                major = nums[i];
                counts = 1;
            } else {
                counts += (nums[i] == major) ? 1 : -1;
            }
        }
        return major;
    }
};